sklearn/ensemble/gradient_boosting.py

```diff
@@ -1396,10 +1440,13 @@ def fit(self, X, y, sample_weight=None, monitor=None):
         # Check input
         X, y = check_X_y(X, y, accept_sparse=['csr', 'csc', 'coo'], dtype=DTYPE)
         n_samples, self.n_features_ = X.shape
+
         if sample_weight is None:
```

Comment:
Nit: Right above this line, we can set `sample_weight_is_none = sample_weight is None` and use it from then on.